rdtools/filtering.py

```diff
@@ -71,56 +89,656 @@ def tcell_filter(temperature_cell, temperature_cell_low=-50,
-    threshold : float, default 0.15
-        threshold for filter
+    power_ac : pandas.Series
+        Pandas time series, representing PV system power or energy.
+        For best performance, timestamps should be in local time.
+    mounting_type: str, default 'fixed'
+        String representing the mounting configuration associated with the
+        AC power or energy time series. Can either be "fixed" or
+        "single_axis_tracking". Default set to 'fixed'.
+    rolling_range_max_cutoff : float, default 0.2
+        Relative fractional cutoff for max rolling range threshold. When the
+        relative maximum range in any interval is below this cutoff, the interval
+        is determined to be clipping. Defaults to 0.2; however, values as high as
+        0.4 have been tested and shown to be effective. The higher the cutoff, the
+        more values in the dataset that will be determined as clipping.
+    roll_periods: int, optional
+        Number of periods to examine when looking for a near-zero derivative
+        in the time series derivative. If roll_periods = 3, the system looks
+        for a near-zero derivative over 3 consecutive readings. Default value
+        is set to None, so the function uses default logic: it looks for a
+        near-zero derivative over 3 periods for a fixed tilt system, and over
+        5 periods for a tracked system with a sampling frequency more frequent
+        than once every 30 minutes.
 
     Returns
     -------
-    pd.Series
-        Boolean Series of whether the clear-sky index is within the threshold
-        around 1.
+    pandas.Series
+        Boolean Series of whether to include the point because it is not
+        clipping.
+        True values delineate non-clipping periods, and False values delineate
+        clipping periods.
+
+    References
+    ----------
+    .. [1] Perry K., Muller, M., and Anderson K. "Performance comparison of clipping
+       detection techniques in AC power time series", 2021 IEEE 48th Photovoltaic
+       Specialists Conference (PVSC). DOI: 10.1109/PVSC43889.2021.9518733.
     '''
+    # Throw a warning that this is still an experimental filter
+    warnings.warn("The logic-based filter is an experimental clipping filter "
+                  "that is still under development. The API, results, and "
+                  "default behaviors may change in future releases (including "
+                  "MINOR and PATCH). Use at your own risk!")
+    # Format the time series
+    power_ac, index_name = _format_clipping_time_series(power_ac,
+                                                        mounting_type)
+    # Test if the data sampling frequency is variable, and flag it if the time
+    # series sampling frequency is less than 95% consistent.
+    _check_data_sampling_frequency(power_ac)
+    # Get the sampling frequency of the time series
+    time_series_sampling_frequency = power_ac.index.to_series().diff()\
```

Comment:
Minor comment: I get the following future warning about timezone-aware datetime series. Also on lines 279 and 653.  Discussion here: https://stackoverflow.com/questions/28565965/pandas-timezone-aware-index-drops-timezone-when-converting-to-series

FutureWarning: The default of the 'keep_tz' keyword will change to True in a future release. You can set 'keep_tz=True' to obtain the future behaviour and silence this warning.
  time_series_sampling_frequency = power_ac.index.to_series().diff()\

Reply:
FYI I'm using Pandas 0.24.2 in this case.  We may not need to change anything here because it looks like if you exclude keep_tz, then you get a warning with pandas 0.24.2 but if you pass keep_tz then you get a warning with Pandas =1.1.3.  So please disregard.

Reply:
Maybe worth using [warnings.catch_warnings](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings) here?  Might as well suppress spurious warnings where we can.

Reply:
I'm inclined not to catch warnings, better to pass them through so users have all the information.

Reply:
I guess I don't see how this warning offers any value to the user.  There's no problem for the user to resolve right?  It's just that pandas is complaining about how we're calling that function, but there's no impact on functionality.  If it doesn't prompt any change on the user's end then to me it seems like useless noise, not "information".  Maybe worse than useless if they spend time trying to fix something they have no control over.

Reply:
Yeah, to follow on Kevin's point @mdeceglie  - we're not going to suppress the warning about the module being experimental, just the useless one about the Pandas.to_series() functionality changing in a future version, which we've already accounted for.  @kanderso-nrel could you provide a pull request that provides the quick fix so that we can have a concrete example to discuss?  Thanks!

Reply:
#291

Reply:
A benefit of not worrying about this is keeping our codebase clean. We can't start policing all possible warnings from all possible combinations of our dependencies. If I understand correctly, the warning is only relevant if someone is using an ancient version of pandas and not relevant to the environment specified in `requirements.txt`.

Reply:
I see the point about keeping the code clean; #291 is pretty cluttered.  Minor quibble about "ancient": pandas 0.25.3 (Nov 1, 2019) has this warning too.  I'm swayed by the requirements.txt argument though... if the user wants a perfect rdtools experience then they should use the recommended environment.  OK with me to leave this as-is.